packages/flutter_tools/lib/src/ios/application_package.dart

```diff
@@ -151,12 +153,34 @@ class BuildableIOSApp extends IOSApp {
       _hostAppBundleName == null ? 'Runner.app' : _hostAppBundleName!,
       'Info.plist');
 
+  // Both project icon's image assets and Contents.json are in the same directory.
+  String get projectAppIconDirName => globals.fs.path.join('ios', _appIconDirNameSuffix);
+
+  // template icon's Contents.json is in flutter_tools.
+  String get templateAppIconDirNameForContentsJson => globals.fs.path.join(
+    Cache.flutterRoot!,
+    'packages/flutter_tools/templates/app_shared/ios.tmpl',
+    _appIconDirNameSuffix,
+  );
+
+  // template icon's image assets are in flutter_template_images package.
+  Future<String> get templateAppIconDirNameForImages async {
+    final Directory imageTemplate = await templateImageDirectory('.', globals.fs, globals.logger);
```

Comment:
Instead of passing `.` I think it would make more sense for `templateImageDirectory` to handle a null name.